addons/mail/models/res_partner.py

```diff
@@ -116,6 +116,7 @@ def im_search(self, name, limit=20):
                 WHERE P.name ILIKE %s
                     AND P.id NOT IN %s
                     AND U.active = 't'
+                    AND U.share IS FALSE
```

Comment:
Actually I think we need to use `IS NOT TRUE` instead of `IS FALSE` to handle potential `null` values correctly. If I'm not mistaken the ORM is considering `null` as `false` for boolean fields, so they should be returned by this query too.